agent/engine/serviceconnect/manager_linux.go

```diff
@@ -372,17 +370,30 @@ func (m *manager) AugmentInstanceContainer(task *apitask.Task, container *apicon
 	return nil
 }
 
-// LoadImage helps load the AppNetAgent container image for the agent
+// LoadImage helps load the AppNetAgent container image for the agent latest supported
+// AppNet interface version by looking for the AppNet agent tar name from supported list
+// of AppNet versions from highest to lowest version when loading AppNet image
 func (agent *manager) LoadImage(ctx context.Context, _ *config.Config, dockerClient dockerapi.DockerClient) (*types.ImageInspect, error) {
-	logger.Debug("Loading appnet agent container tarball:", logger.Fields{
-		field.Image: agent.AgentContainerTarballPath,
-	})
-	if err := loader.LoadFromFile(ctx, agent.AgentContainerTarballPath, dockerClient); err != nil {
-		return nil, err
+	var err error
+	for _, supportedAppnetInterfaceVersion := range getSupportedAppnetInterfaceVersions() {
+		agentContainerTarballPath := fmt.Sprintf(defaultAgentContainerTarballPathFormat, supportedAppnetInterfaceVersion)
+		if _, err = os.Stat(agentContainerTarballPath); err == nil {
+			logger.Debug("Loading appnet agent container tarball:", logger.Fields{
+				field.Image: agentContainerTarballPath,
+			})
+			if loadErr := loader.LoadFromFile(ctx, agentContainerTarballPath, dockerClient); err != nil {
+				logger.Warn("Unable to load appnet agent container tarball:", logger.Fields{
+					field.Image: agentContainerTarballPath,
+					field.Error: loadErr,
+				})
+				continue
+			}
+			agent.AppnetInterfaceVersion = supportedAppnetInterfaceVersion
+			imageName, _ := agent.GetLoadedImageName()
+			return loader.GetContainerImage(imageName, dockerClient)
```

Comment:
Let's log an info message here to state the tar we've successfully loaded, and what we are assuming the loaded image name to be

Reply:
Added the info message.